taichi/ir/ir_builder.cpp

```diff
@@ -44,6 +44,38 @@ void IRBuilder::set_insertion_point_to_false_branch(IfStmt *if_stmt) {
   set_insertion_point({if_stmt->false_statements.get(), 0});
 }
 
+IRBuilder::LoopGuard::~LoopGuard() {
+  if (location_ >= 0 && location_ < loop_->parent->size() &&
```

Comment:
nit: create a helper for loop and if, e.g., `stmt_location_did_not_change(stmt, location)`?